src/mono/wasm/runtime/jiterpreter.ts

```diff
@@ -1993,6 +2007,22 @@ function emit_binop (builder: WasmBuilder, ip: MintOpcodePtr, opcode: MintOpcode
         info : OpRec3 | OpRec4 | undefined,
         operandsCached = false;
 
+    const intrinsicFpBinop = intrinsicFpBinops[opcode];
+    if (intrinsicFpBinop) {
+        builder.local("pLocals");
+        const isF64 = intrinsicFpBinop == WasmOpcode.nop;
+        append_ldloc(builder, getArgU16(ip, 2), isF64 ? WasmOpcode.f64_load : WasmOpcode.f32_load);
+        if (!isF64)
+            builder.appendU8(intrinsicFpBinop);
+        append_ldloc(builder, getArgU16(ip, 3), isF64 ? WasmOpcode.f64_load : WasmOpcode.f32_load);
+        if (!isF64)
+            builder.appendU8(intrinsicFpBinop);
+        builder.i32_const(<any>opcode);
+        builder.callImport("relop_fp");
```

Comment:
Do I understand correctly this is call from generated wasm into main wasm file ? is this expensive ?

Reply:
Yes, it's calling into dotnet.wasm - in this case, mono_jiterp_relop_fp. It is somewhat expensive, but the alternative is painful. In the long run if the performance of these comparisons was sensitive we'd generate all the ordering comparison logic inline.